juice/src/layers/common/rnn.rs

```diff
@@ -67,29 +64,120 @@ pub enum RnnType {
     tanh,
 }
 
-impl RnnType {
+impl RnnNetworkMode_UserInput {
```

Comment:
I am not really sold on the duplication, the ones in `plugin.rs` should be good across all backend impls (so you can share the configs)

Reply:
Me neither. I was replicating RnnType and thought it'd be the best way of doing it to have a mirror in Juice. 

I think you're right, I can add in to/from/display in plugin for the text so it'll deal with user input, and strip them out of here.